tools/doc/html.mjs

```diff
@@ -205,6 +205,65 @@ function linkJsTypeDocs(text) {
 }
 
 const isJSFlavorSnippet = (node) => node.lang === 'cjs' || node.lang === 'mjs';
+const copyButton = '<button class="copy-button">copy</button>';
+
+function processJSFlavorSnippet(node, filename, index, parent) {
+  const previousNode = parent.children[index - 1] || {};
+  const nextNode = parent.children[index + 1] || {};
+  const highlighted = `<code class='language-js ${node.lang}'>${(getLanguage(node.lang || '') ? highlight(node.value, { language: node.lang }) : node).value}</code>`;
+  const charCountFirstTwoLines = Math.max(...node.value.split('\n', 2).map((str) => str.length));
+
+  if (!isJSFlavorSnippet(previousNode) &&
+    isJSFlavorSnippet(nextNode) &&
+    nextNode.lang !== node.lang) {
+    // Saving the highlight code as value to be added in the next node.
+    node.value = highlighted;
+    node.charCountFirstTwoLines = charCountFirstTwoLines;
+  } else if (isJSFlavorSnippet(previousNode) &&
+    previousNode.lang !== node.lang) {
+    const actualCharCount = Math.max(charCountFirstTwoLines, previousNode.charCountFirstTwoLines);
+    (dynamicSizes[filename] ??= new Set()).add(actualCharCount);
+    node.value = `<pre class="with-${actualCharCount}-chars">` +
+      '<input class="js-flavor-toggle" type="checkbox"' +
+      // If CJS comes in second, ESM should display by default.
+      (node.lang === 'cjs' ? ' checked' : '') +
+      ' aria-label="Show modern ES modules syntax">' +
+      previousNode.value +
+      highlighted +
+      copyButton +
+      '</pre>';
+    node.lang = null;
+    previousNode.value = '';
+    previousNode.lang = null;
+  } else {
+    // Isolated JS snippet, no need to add the checkbox.
+    node.value = `<pre>${highlighted} ${copyButton}</pre>`;
+  }
+}
+
+function toCJS(value) {
+  // Find import comments '//@import '{ Buffer }' 'node:buffer'
+  const importComment = /\/\/@import\s*'([^']+?)'\s*'([^']+?)'[^\n]*/g;
```

Comment:
Without additional regexp flags, will this match multi-line blocks? For example:
```js
//@import '{
  Buffer,
  File,
}' 'node:buffer';
```